src/device/usbd.c

```diff
@@ -618,7 +618,6 @@ static bool process_control_request(uint8_t rhport, tusb_control_request_t const
       TU_ASSERT(ep_num < TU_ARRAY_SIZE(_usbd_dev.ep2drv) );
 
       uint8_t const drvid = _usbd_dev.ep2drv[ep_num][ep_dir];
-      TU_ASSERT(drvid < USBD_CLASS_DRIVER_COUNT);

```

Comment:
Yeah, this maybe possible. But I am curious which request the macos use to run into this issue. Since the only request targeted endpoint is halt/clear halt. Which will never be used on Control Endpoint anyway

Reply:
It’s a GET_STATUS on EP0.

Reply:
According to the USB2.0 spec, this is definitely valid. In fact, in the address state the *only* valid GET_STATUS endpoint is EP0. See section 9.4.5. The last paragraph also basically says that GET_STATUS should never stall: 

“It is neither required nor recommended that the Halt feature be implemented for the Default Control Pipe. However, devices may set the Halt feature of the Default Control Pipe in order to reflect a functional error condition.  If the feature is set to one, the device will return STALL in the Data and Status stages of each standard request to the pipe **except GetStatus()**, SetFeature(), and ClearFeature() requests.”

Reply:
Right, right, I forgot the  the GET_STATUS request, these are rarely used on control endpoints. Thank you very much for the explanation.